core/src/main/java/org/dozer/config/processors/DefaultSettingsProcessor.java

```diff
@@ -0,0 +1,93 @@
+
+    public DefaultSettingsProcessor(DozerClassLoader classLoader) {
+        this.classLoader = classLoader;
+    }
+
+    @Override
+    public Settings process() {
+        SettingsResolver settingsResolver = getSettingsResolver();
+        if (settingsResolver.get(Settings.Keys.DEPRECATED_CLASS_LOADER_BEAN, null) != null) {
+            throw new IllegalArgumentException("Found key in properties for " + Settings.Keys.DEPRECATED_CLASS_LOADER_BEAN
+                                               + ". This is deprecated, please use: "
+                                               + Settings.Keys.CLASS_LOADER_BEAN);
+        }
+
+        if (settingsResolver.get(Settings.Keys.DEPRECATED_PROXY_RESOLVER_BEAN, null) != null) {
+            throw new IllegalArgumentException("Found key in properties for " + Settings.Keys.DEPRECATED_PROXY_RESOLVER_BEAN
+                                               + ". This is deprecated, please use: "
+                                               + Settings.Keys.PROXY_RESOLVER_BEAN);
+        }
+
+        Boolean elEnabled = Boolean.valueOf(settingsResolver.get(Settings.Keys.EL_ENABLED, DozerConstants.DEFAULT_EL_ENABLED).toString());
+        String classLoaderBeanName = String.valueOf(settingsResolver.get(Settings.Keys.CLASS_LOADER_BEAN, DozerConstants.DEFAULT_CLASS_LOADER_BEAN));
+        String proxyResolverBeanName = String.valueOf(settingsResolver.get(Settings.Keys.PROXY_RESOLVER_BEAN, DozerConstants.DEFAULT_PROXY_RESOLVER_BEAN));
+
+        Integer converterByDestTypeCacheMaxSize = Integer.valueOf(settingsResolver.get(Settings.Keys.CONVERTER_BY_DEST_TYPE_CACHE_MAX_SIZE,
+                                                                                       DozerConstants.DEFAULT_CONVERTER_BY_DEST_TYPE_CACHE_MAX_SIZE).toString());
+
+        Integer superTypesCacheMaxSize = Integer.valueOf(settingsResolver.get(Settings.Keys.SUPER_TYPE_CHECK_CACHE_MAX_SIZE,
+                                                                              DozerConstants.DEFAULT_SUPER_TYPE_CHECK_CACHE_MAX_SIZE).toString());
+
+        Settings answer = new Settings();
+        answer.setConverterByDestTypeCacheMaxSize(converterByDestTypeCacheMaxSize);
+        answer.setSuperTypesCacheMaxSize(superTypesCacheMaxSize);
+        answer.setElEnabled(elEnabled);
+        answer.setClassLoaderBeanName(classLoaderBeanName);
+        answer.setProxyResolverBeanName(proxyResolverBeanName);
+
+        return answer;
+    }
+
+    private SettingsResolver getSettingsResolver() {
+        String configFileNameFound = System.getProperty(DozerConstants.CONFIG_FILE_SYS_PROP);
+        if (MappingUtils.isBlankOrNull(configFileNameFound)) {
+            configFileNameFound = DozerConstants.DEFAULT_CONFIG_FILE;
+        }
+
+        SettingsResolver defaultValueSettingsResolver = new DefaultValueSettingsResolver();
+        SettingsResolver systemEnvironmentSettingsResolver = new SystemEnvironmentSettingsResolver(defaultValueSettingsResolver);
```

Comment:
Another alternative for implementing chained resolvers would be to have a "composite" resolver which iterates over all registered resolvers and returns the first non-null value. The benefit would be that we do not need to bother with `fallthrough` in every implementation.